Lib/modulefinder.py

```diff
@@ -43,6 +51,67 @@ def ReplacePackage(oldname, newname):
     replacePackageMap[oldname] = newname
 
 
+def _find_module(name, path=None):
+    """An importlib reimplementation of imp.find_module (for our purposes)."""
+
+    importlib.machinery.PathFinder.invalidate_caches()
+
+    spec = importlib.machinery.PathFinder.find_spec(name, path+sys.path)
+
+    if spec is None:
+        raise ImportError("No module named {name!r}".format(name=name), name=name)
+
+    if isinstance(spec.loader, type):  # Some special cases:
```

Comment:
Not sure if using ``issubclass`` necessary, just testing for these two special cases is good enough:

```
if spec.loader is import lib.machinery.BuiltinImporter:
  ...
```

Another special case is the loader for implicit namespace packages (which is currently an instance of a private class, see bpo-35673). Those are probably not handled at all at the moment by this module.

Reply:
I've refactored these to use identity checking, as you suggested. Much cleaner.

I think it makes the most sense to wait on bpo-35673's resolution before trying to hack our way around namespace packages here... that could be a separate PR. What are your thoughts?